snuba/cli/subscriptions.py

```diff
@@ -167,43 +168,45 @@ def subscriptions(
     metrics.gauge("executor.workers", executor._max_workers)
 
     with closing(consumer), executor, closing(producer):
-        batching_consumer = BatchingConsumer(
+        batching_consumer = StreamProcessor(
             consumer,
             (
                 Topic(topic)
                 if topic is not None
                 else Topic(loader.get_default_topic_spec().topic_name)
             ),
-            SubscriptionWorker(
-                dataset,
-                executor,
-                {
-                    index: SubscriptionScheduler(
-                        RedisSubscriptionDataStore(
-                            redis_client, dataset, PartitionId(index)
-                        ),
-                        PartitionId(index),
-                        cache_ttl=timedelta(seconds=schedule_ttl),
-                        metrics=metrics,
-                    )
-                    for index in range(
-                        partitions
-                        if partitions is not None
-                        else loader.get_default_topic_spec().partitions_number
-                    )
-                },
-                producer,
-                Topic(result_topic),
-                metrics,
-                time_shift=(
-                    timedelta(seconds=delay_seconds * -1)
-                    if delay_seconds is not None
-                    else None
+            BatchProcessorFactory(
+                SubscriptionWorker(
+                    dataset,
+                    executor,
+                    {
+                        index: SubscriptionScheduler(
+                            RedisSubscriptionDataStore(
+                                redis_client, dataset, PartitionId(index)
+                            ),
+                            PartitionId(index),
+                            cache_ttl=timedelta(seconds=schedule_ttl),
+                            metrics=metrics,
+                        )
+                        for index in range(
+                            partitions
+                            if partitions is not None
+                            else loader.get_default_topic_spec().partitions_number
+                        )
+                    },
+                    producer,
+                    Topic(result_topic),
+                    metrics,
+                    time_shift=(
+                        timedelta(seconds=delay_seconds * -1)
+                        if delay_seconds is not None
+                        else None
+                    ),
                 ),
+                max_batch_size,
+                max_batch_time_ms,
+                metrics,
             ),
```

Comment:
This indentation level is kind of ridiculous but hopefully will be able to move the subscription code out of the batch processor soon.